crates/store/re_grpc_client/src/lib.rs

```diff
@@ -71,26 +71,62 @@ enum StreamError {
 
 // ----------------------------------------------------------------------------
 
+/// Stream recordings catalog over gRPC from Rerun Data Platform.
+///
+/// `on_msg` can be used to wake up the UI thread on Wasm.
+pub fn stream_catalog(
+    redap_url: url::Url,
+    on_msg: Option<Box<dyn Fn() + Send + Sync>>,
+) -> Result<re_smart_channel::Receiver<LogMsg>, InvalidRedapAddress> {
+    re_log::debug!("Loading {redap_url}…");
+
+    let address = redap_url.clone().try_into()?;
+
+    let (tx, rx) = re_smart_channel::smart_channel(
+        re_smart_channel::SmartMessageSource::RerunGrpcStream {
+            url: redap_url.clone().to_string(),
+        },
+        re_smart_channel::SmartChannelSource::RerunGrpcStream {
+            url: redap_url.clone().to_string(),
+        },
+    );
+
+    spawn_future(async move {
+        if let Err(err) = stream_catalog_async(tx, address, on_msg).await {
+            re_log::warn!(
+                "Error while streaming {redap_url}: {}",
+                re_error::format_ref(&err)
+            );
+        }
+    });
+
+    Ok(rx)
+}
+
 /// Stream an rrd file over gRPC from a Rerun Data Platform server.
 ///
 /// `on_msg` can be used to wake up the UI thread on Wasm.
 pub fn stream_recording(
-    url: String,
+    redap_url: Url,
     on_msg: Option<Box<dyn Fn() + Send + Sync>>,
-) -> Result<re_smart_channel::Receiver<LogMsg>, InvalidAddressError> {
-    re_log::debug!("Loading {url}…");
+) -> Result<re_smart_channel::Receiver<LogMsg>, InvalidRedapAddress> {
+    re_log::debug!("Loading {redap_url}…");
 
-    let address = Address::from_str(&url)?;
+    let address = redap_url.clone().try_into()?;
 
     let (tx, rx) = re_smart_channel::smart_channel(
-        re_smart_channel::SmartMessageSource::RerunGrpcStream { url: url.clone() },
-        re_smart_channel::SmartChannelSource::RerunGrpcStream { url: url.clone() },
+        re_smart_channel::SmartMessageSource::RerunGrpcStream {
+            url: redap_url.clone().to_string(),
+        },
+        re_smart_channel::SmartChannelSource::RerunGrpcStream {
+            url: redap_url.clone().to_string(),
+        },
     );
 
     spawn_future(async move {
         if let Err(err) = stream_recording_async(tx, address, on_msg).await {
             re_log::warn!(
-                "Error while streaming {url}: {}",
+                "Error while streaming {redap_url}: {}",
                 re_error::format_ref(&err)
             );
```

Comment:
This is where I think we could put a match statement on the new enum `RedapAddress` and split between `stream_recording_async` and `stream_catalog_async` depending on the match arm.